client/state/index.js

```diff
@@ -128,7 +128,8 @@ export function createReduxStore( initialState = {} ) {
 		enhancers.push( sitesSync );
 
 		if ( window.app && window.app.isDebug ) {
-			enhancers.push( consoleDispatcher );
+			// this has to be the first enhancer in the chain for the other enhancers to see it
+			enhancers.unshift( consoleDispatcher );
```

Comment:
> How does that first filter invocation work?

oops. it doesn't. :smile: this is why I tell people to never copy and paste my code from review comments. it was stream-of-thought coding.

```js
const enhancers = [
	…
].filter( ( [ first, second ] ) => !! second ).map( first )
```

alternatively as a reducer

```js
const enhancers = [
	…
].reduce( ( [ finalChain, [ enhancer, shouldInclude ] ) =>
	shouldInclude
		? finalChain.concat( enhancer )
		: finalChain
, [] );
```